tiledb/sm/query/query_condition.cc

```diff
@@ -1473,45 +1523,163 @@ Status QueryCondition::apply_clause_sparse(
       return Status_QueryConditionError(
           "Cannot perform query comparison; Unsupported query "
           "conditional type on " +
-          clause.field_name_);
+          node->field_name_);
   }
 
   return Status::Ok();
 }
 
+template <typename BitmapType>
+void QueryCondition::apply_tree_sparse(
+    const tdb_unique_ptr<ASTNode>& node,
+    const ArraySchema& array_schema,
+    ResultTile& result_tile,
+    std::vector<BitmapType>& result_bitmap) const {
```

Comment:
Please add more documentation to describe what each block of code is trying to accomplish.